app/admin/top-issues/components/TopIssuesList.js

```diff
@@ -95,109 +97,119 @@ export default function TopIssuesList({ topIssues }) {
       };
     });
     await deleteTopIssueCallback(showIssueDeleteAlert);
+    setTopIssues(topIssues.filter(({id}) => id !== showIssueDeleteAlert))
     setShowIssueDeleteAlert(false);
-    window.location.reload();
   };
+
   return (
     <div>
-      {topIssues.map((issue) => (
-        <div className="py-3 mb-3 border-t border-t-stone-500" key={issue.id}>
-          <div className="flex items-center justify-between">
-            <div className="flex items-center">
-              <GoIssueOpened />
-              <strong>&nbsp; {issue.name}</strong>
-            </div>
-            <div className="flex items-center">
-              <BlackButtonClient
-                onClick={() => {
-                  setAddNewPosition(issue.id);
-                }}
-              >
-                <strong>Add a position for {issue.name}</strong>
-              </BlackButtonClient>{' '}
-              <div
-                className="text-red-600 inline-block ml-4 bg-stone-300 rounded-full p-4 w-12 h-12 text-center cursor-pointer"
-                onClick={() => {
-                  setShowIssueDeleteAlert(issue.id);
-                }}
-              >
-                <FaTrash />
+      {
+        topIssues.map((issue) => (
+          <div className="py-3 mb-3 border-t border-t-stone-500" key={issue.id}>
+            <div className="flex items-center justify-between">
+              <div className="flex items-center">
+                <TopIssueDisplay {...{
+                  issue,
+                  topIssues,
+                  setTopIssues
+                }} />
               </div>
-            </div>
-          </div>
-          {addNewPosition === issue.id && (
-            <div>
-              <br />
-              <br />
-              <TextField
-                fullWidth
-                primary
-                label="Position Name"
-                value={positionName}
-                onChange={(e) => setPositionName(e.target.value)}
-              />
-              <br />
-              <br />
-              <div className="text-right">
+              <div className="flex items-center">
                 <BlackButtonClient
```

Comment:
change to primaryButton and moe the onClick event to the parent div

Reply:
Doesn't that `div` include the trash/delete button?

You want the `onClick` for adding a new issue position to be on the wrapper `div` that also holds the trash/delete button? 🤔 

Also. I didn't modify this code w/r/t the `BlackButtonClient` use.  I think it's just marking these lines as "touched" because they moved when I abstracted out `TopIssueDisplay` in the lines proceeding these buttons.

Reply:
https://github.com/thegoodparty/gp-webapp/pull/122/commits/8d711b2745dc7d7fcfa7eab514c9588ec831cbd5